### PR TITLE
etcdclientv3: setup initial structure

### PR DESCRIPTION
**Pull request by xiang90**
Set up initial client structure so that we can iterate on fast.

### CODE REVIEW

File: clientv3/kv.go [hunk 1]
```diff
@@ -14,27 +14,32 @@
...
+
 type KV interface {
 	// PUT puts a key-value pair into etcd.
 	// Note that key,value can be plain bytes array and string is
 	// an immutable representation of that bytes array.
 	// To get a string of bytes, do string([]byte(0x10, 0x20)).
-	Put(key, val string) (PutResponse, error)
+	Put(key, val string) (*pb.PutResponse, error)
 
 	// Range gets the keys [key, end) in the range at rev.
```
Comment (heyitsanthony) [inline clientv3/kv.go]: `type RangeReponse pb.RangeResponse` instead of exposing protobuf type?

File: clientv3/op.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	// A default Op has opType 0, which is invalid.
+	tRange       = opType(1)
+	tPut         = opType(2)
+	tDeleteRange = opType(3)
+)
+
+// Op represents an Operation that kv can execute.
+type Op struct {
+	t   opType
+	key []byte
```
Comment (heyitsanthony) [inline clientv3/op.go]: why the switch from `string` to `[]byte`?

File: clientv3/kv.go (hunk 1 above)
Comment (xiang90) [inline clientv3/kv.go]: good ideal. I really do not want to expose pb. :)

File: clientv3/kv.go [hunk 3]
```diff
@@ -67,33 +72,34 @@ type Txn interface {
...
-
-type SortTarget int
-type SortOrder int
-
-type SortOption struct {
-	Target SortTarget
-	Order  SortOrder
+func (kv *kv) do(op Op) *pb.ResponseUnion {
+	switch op.t {
+	// TODO: handle other ops
```
Comment (heyitsanthony) [inline clientv3/kv.go]: `default: panic()`

File: clientv3/op.go (hunk 2 above)
Comment (xiang90) [inline clientv3/op.go]: 1. pb struct requires []byte anyway
2. differentiate nil from empty so that we do not need tGet type

Comment (heyitsanthony) [inline clientv3/op.go]: ok

File: clientv3/op.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clientv3
+
+type opType int
+
+var (
+	// A default Op has opType 0, which is invalid.
+	tRange       = opType(1)
```
Comment (heyitsanthony) [inline clientv3/op.go]: `iota`?

File: clientv3/kv.go [hunk 5]
```diff
@@ -67,33 +80,36 @@ type Txn interface {
 	Else(ops ...Op) Txn
 
 	// Commit tries to commit the transaction.
-	Commit() (TxnResponse, error)
+	Commit() (*pb.TxnResponse, error)
```
Comment (heyitsanthony) [inline clientv3/kv.go]: missed this one

**Comment by xiang90** [comment]
@heyitsanthony All fixed. Not sure why travis failed. Seems like a travis issue.

File: clientv3/op.go [hunk 6]
```diff
@@ -0,0 +1,72 @@
+// Copyright 2015 CoreOS, Inc.
```
Comment (heyitsanthony) [inline clientv3/op.go]: 2016?

Comment (xiang90) [inline clientv3/op.go]: fixed.

**Comment by heyitsanthony** [comment]
lgtm

**Pull request merged**